snuba/query/parser/__init__.py

```diff
@@ -275,7 +281,7 @@ def _validate_empty_table_names(query: Query) -> None:
         )
 
 
-def _validate_aliases(query: Query) -> None:
+def _validate_aliases(query: Union[CompositeQuery[QueryEntity], Query]) -> None:
```

Comment:
I like the idea of having these functions be specific to a query, and letting an outer wrapping function be the one that recurses down through the subqueries. I didn't add anything to the current parser because I wasn't sure how the subqueries are being created/used right now.